### PR TITLE
Add target=_blank to element links 

### LINKED ISSUE #4719: Editor: ensure links open in new window
## Feature Description

Only `target=_blank` links are valid in `amp-story`.

Right now we don't add the `target` attribute at all, which surprisingly works.

See #4689.

See #4705

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by dreamofabear**
A copy of #4705 but from a branch rather than a fork.

## Summary

Adds attribute `target=_blank` (and `rel=noreferrer`) to element links.

## Relevant Technical Choices

- Only `target=_blank` links are valid in `amp-story`.
- Avoids `target=_top` being added by AMP packager v2 (bug).

## To-do

N/A

## User-facing changes

N/A

## Testing Instructions

- Add a link to an image or text element and publish the story.
- Inspect the `<a>` tag and verify that it has attribute `target="_blank"`.

---

Partial for #4689.

Fixes #4719

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
> A copy of #4705 but from a branch rather than a fork.

Doesn't matter too much, but alas, makes lives a bit easier for now.

Note: don't merge before QA passed

**Comment by dreamofabear** [comment]
> Note: don't merge before QA passed

Gotcha. I don't have merge rights anyways. 🙃

File: assets/src/edit-story/components/elementLink/test/output.js
Comment (barklund) [inline assets/src/edit-story/components/elementLink/test/output.js]: What's the difference between these two test cases?

Comment (swissspidy) [reply assets/src/edit-story/components/elementLink/test/output.js]: This test (`should produce valid AMP output for one-tap links`) can actually be removed. We don't have one-tap links anymore.

That also means removing `type: 2,` from the `props` above.

**Pull request merged**
